service/src/test/java/greencity/filters/HabitFactSpecificationTest.java

```diff
@@ -0,0 +1,236 @@
+package greencity.filters;
+
+import greencity.entity.*;
+import jakarta.persistence.criteria.*;
+import jakarta.persistence.metamodel.SingularAttribute;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+
+@ExtendWith(MockitoExtension.class)
+class HabitFactSpecificationTest {
+
+    @Mock
+    private Root<HabitFact> HabitFactRoot;
+
+    @Mock
+    private CriteriaBuilder builder;
+
+    @Mock
+    private CriteriaQuery MockCriteriaQuery;
+
+    @Mock
+    private Path<Object> id;
+
+    @Mock
+    private Path<Long> habitIdPath;
+
+    @Mock
+    private Path<Long> habitFactRootIdPath;
+
+    @Mock
+    private Join<HabitFact,Habit> habitJoin;
+
+    @Mock
+    private Path<String> contentPath;
+
+    @Mock
+    private Predicate likePredicate;
+
+    @Mock
+    private Expression<String> as;
+    HabitFactSpecification specification;
```

Comment:
for some predicates/paths/etc you use local variables, but for some others you use fields; maybe use local variables for everything, or at least document which fields correspond to which test?